### PR TITLE
flow fragment support shapex.tie_shape argument outside of the fragment body

### PR DESCRIPTION
**Pull request by shanshanpt**
flow fragment support shapex.tie_shape argument outside of the fragment body

### CODE REVIEW

**Comment by shanshanpt** [comment]
fixed #2729

File: iree/compiler/Dialect/Flow/Transforms/test/form_streams.mlir [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+  // CHECK-NEXT: cmpi "slt", %11, %4 : i32
+  %12 = cmpi "slt", %11, %4 : i32
+  // CHECK-NEXT: select %12, %11, %4 : i32
+  %13 = select %12, %11, %4 : i32
+  // CHECK-NEXT: %14 = shapex.ranked_dim %arg1[0] : !shapex.ranked_shape<[?]> -> index
+  // CHECK-NEXT: %15 = shapex.ranked_dim %1[0] : !shapex.ranked_shape<[?]> -> index
+  // CHECK-NEXT: %16 = flow.ex.stream.fragment(%arg2 = %arg0 : tensor<?xi32>, %arg3 = %14 : index, %arg4 = %[[WORKLOAD3]] : index, %arg5 = %8 : i32, %arg6 = %13 : i32, %arg7 = %[[WORKLOAD1]] : i32, %arg8 = %[[WORKLOAD2]] : index, %arg9 = %2 : tensor<?xi32>, %arg10 = %0 : index, %arg11 = %15 : index) -> tensor<?xi32> {
+  // CHECK-NEXT:   %17 = shapex.make_ranked_shape %arg3 : (index) -> !shapex.ranked_shape<[?]>
+  // CHECK-NEXT:   %18 = shapex.make_ranked_shape %arg11 : (index) -> !shapex.ranked_shape<[?]>
+  // CHECK-NEXT:   %19 = shapex.tie_shape %arg2, %17 : tensor<?xi32>, !shapex.ranked_shape<[?]>
```
Comment (stellaraccident) [inline iree/compiler/Dialect/Flow/Transforms/test/form_streams.mlir:46]: The presence of this op is really what the test case is trying to verify, right? Could we please reduce the test case to be much less fragile in terms of CHECKs, relying on general tests for the other invariants? For example, other parts of the infra/tests verify that capture and mapping logic is correct, and the verifier will fail if it is mechanically done wrong. I think I would be fine with just a check that a stream gets produced with a tie_shape that feeds into the proper flow.dispatch (by name). Removing all of the other over-constrained checks will make this more maintainable and robust to incidental changes.

**Comment by stellaraccident** [comment]
Thanks. This test is still a bit more fixed than feels right. But it is fine to submit: I've added it to my task list to see if I can simplify it a bit more next week. Thanks for the fix!

**Comment by shanshanpt** [comment]
> Thanks. This test is still a bit more fixed than feels right. But it is fine to submit: I've added it to my task list to see if I can simplify it a bit more next week. Thanks for the fix!

OK，thanks.

**Pull request merged**